app/sockets/service.py

```diff
@@ -101,7 +101,44 @@ def create_conversation(conversation_id, creator, participants):
                 result.append(participant)
                 
         return result
-    
+
+    @staticmethod
+    def remove_identifier(email):
+        """
+        Removes the provided email from the internal mapping
+        of users to socket IDs
+        
+        :email        the email address of a user
+        """
+        
+        if email in SocketService._sockets:
+            del SocketService._sockets[email]
+
+    @staticmethod
+    def leave_conversation(socket_id, conversation_id, participant):
```

Comment:
The caller ultimately should not need to know the `socket_id` or `conversation_id` in order to leave the conversation. Only `participant` is really needed.